pyiron_base/job/script.py

```diff
@@ -31,6 +31,99 @@ class ScriptJob(GenericJob):
+                ```
+
+        Step 3. Check the job table to get details about 'scriptjob' by using:
+                ```
+                pr.job_table()
+                ```
+
+        Step 4. If the status of 'scriptjob' is 'finished', load the data from the JSON file into the
+                'submit_example_job.ipynb' notebook by using:
+                ```
+                import json
+                with open(scriptjob.working_directory + '/script_output.json') as f:
+                    data = json.load(f)  # load the data from the JSON file
+                ```
+
+    More sophisticated example:
+        The script in ScriptJob can also be more complex, e.g. running its own pyiron calculations.
+        Here we show how it is leveraged to run a multi-core atomistic calculation.
+
+        Step 1. 'example.ipynb' can contain pyiron_atomistics code like:
+                ```
+                from pyiron_atomistics import Project
+                pr = Project('example')
+                job = pr.create.job.Lammps('job')  # we name the job 'job'
+                job.structure = pr.create.structure.ase_bulk('Fe')  # specify structure
+
+                # Optional: get an input value from 'submit_example_job.ipynb', the notebook which submits
+                #   'example.ipynb'
+                external_input = pr.get_external_input()
+                job.server.cores = external_input['number_of_cores']
+
+                job.run()  # run the job
+
+                # save a custom output, that can be used by the notebook 'submit_example_job.ipynb'
+                job['user/my_custom_output'] = 16
+                ```
+
+        Step 2. 'submit_example_job.ipynb', can then have the following code:
+                ```
+                from pyiron_base import Project
+                pr = Project('scriptjob_example')  # save the ScriptJob in the 'scriptjob_example' project
+                scriptjob = pr.create.job.ScriptJob('scriptjob')  # create a ScriptJob named 'scriptjob'
+                scriptjob.script_path = 'example.ipynb'  # specify the PATH to the notebook you want to submit.
+                                                         # In this example case, 'example.ipynb' is in the same
+                                                         # directory as 'submit_example_job.ipynb'
+
+                # Optional: to submit the notebook to a queueing system
+                number_of_cores = 4  # number of cores to be used
+                scriptjob.server.cores = number_of_cores
+                scriptjob.server.queue = 'cmfe'  # specify the queue to which the ScriptJob job is to be submitted
+                scriptjob.server.run_time = 120  # specify the runtime limit for the ScriptJob job in seconds
+
+                # Optional: save an input, such that it can be accessed by 'example.ipynb'
+                scriptjob.input['number_of_cores'] = number_of_cores
```

Comment:
Also here can save data at the project level for use in script jobs with `pr.data.number_of_cores = number_of_cores; pr.data.write()` as of #192

Reply:
and done.

Reply:
@liamhuber if this looks good, shall I merge this pull request?

Reply:
@raynol-dsouza I guess wait until the end of the week, by then Jan should have had a chance to review #192 and we can merge both.

Reply:
@liamhuber I accidentally merged it before you posted this comment. O_O'

I will revert the pull request if further changes to #192 affect this PR.